types/time.go

```diff
@@ -1020,7 +1183,20 @@ type Duration struct {
 	Fsp int8
 }
 
-//Add adds d to d, returns a duration value.
+// MaxMySQLDuration returns Duration with maximum mysql time.
+func MaxMySQLDuration(fsp int8) Duration {
+	return NewDuration(TimeMaxHour, TimeMaxMinute, TimeMaxSecond, 0, fsp)
+}
+
+// Neg negative d, reutrns a duration value.
```

Comment:
```suggestion
// Neg negative d, returns a duration value.
```

Reply:
And what the meaning of negative d?  What will return if I call `d.Neg().Hour()`? What the meaning of the value?
If the function is only used by `TestMixDateAndTime`, I think we can use `mixDateAndDuration(date *Time, dur Duration, neg bool)` instead of `mixDateAndDuration(date *Time, dur Duration)` and remove this function.

Reply:
It seems that mixDateAndDuration can receive `gotime.Duration`

Reply:
In fact that Duration can be negative, so `mixDateAndDuration` accept a neg flag is confusing.

The gotime.Duration can also be negative, and `(-d).Hour()` will also produce a positive int. The behaviour of `Neg` is exactly as `-` in gotime.Duration.

Reply:
The question is that we lack a method to create a negative duration quickly. One solution as add the flag to `NewDuration`, or add a new method `NewNegDuration`

Reply:
And because negative Duration already exists, add a `Neg` method to simplify creating is reasonable.

Reply:
OK. I think you are right. And I will review this PR again.